src/utils/azureUtils.ts

```diff
@@ -22,18 +25,100 @@ export function getResourceGroupFromId(id: string): string {
     return parseResourceId(id)[2];
 }
 
-export function createGroupConfigFromResource(resource: GenericResource): GroupingConfig {
+export function createGroupConfigFromResource(resource: GenericResource, subscriptionId: string | undefined): GroupingConfig {
     const id = nonNullProp(resource, 'id');
     return {
         resourceGroup: { keyLabel: 'Resource Groups', label: getResourceGroupFromId(id), id: id.substring(0, id.indexOf('/providers')).toLowerCase() },
         resourceType: {
-            keyLabel: 'Resource Types', label: resource.type?.toLowerCase() || 'unknown',
-            id: `${id.substring(0, id.indexOf('/resourceGroups'))}/${resource.type}` || 'unknown'
+            keyLabel: 'Resource Types', label: resource.type?.toLowerCase() ?? 'unknown',
+            id: `${subscriptionId}/${resource.type}` || 'unknown',
+            iconPath: getIconPath(resource?.type || 'resource')
```

Comment:
Yeah, sorry, `||` is just an old habit because `??` didn't exist in the past.  I agree that we should be explicit and use `??` since it's safer (unless we want to defer to the next value for falsey values as well)